x-pack/plugin/data-streams/src/main/java/org/elasticsearch/xpack/datastreams/mapper/DataTierFieldMapper.java

```diff
@@ -0,0 +1,97 @@
+/*
```

Comment:
I am not sure this mapper should be added in the datastreams module. For now we could add it in x-pack core next to the DataTier definition ?